applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp

```diff
@@ -394,20 +394,22 @@ void NavierStokesWallCondition<TDim,TNumNodes>::ComputeRHSNeumannContribution(ar
 
 
 template<unsigned int TDim, unsigned int TNumNodes>
-void NavierStokesWallCondition<TDim,TNumNodes>::ComputeRHSOutletInflowContribution(array_1d<double,TNumNodes*(TDim+1)>& rhs_gauss,
-                                                                                   const ConditionDataStruct& data)
+void NavierStokesWallCondition<TDim,TNumNodes>::ComputeRHSOutletInflowContribution(
+    array_1d<double,TNumNodes*(TDim+1)>& rhs_gauss,
+    const ConditionDataStruct& data)
 {
     const unsigned int LocalSize = TDim+1;
     const GeometryType& rGeom = this->GetGeometry();
 
+    // Get DENSITY from parent element properties
+    auto & r_neighbours = this->GetValue(NEIGHBOUR_ELEMENTS);
+    const double rho = r_neighbours[0].GetProperties().GetValue(DENSITY);
```

Comment:
Just for curiosity, are we gonna do it like this or are we gonna assign same properties object(which is used on elements) to conditions when creating/replacing them?

Since these are called once per local matrix/vector build, I don't know the implications on computational cost (may be not too significant?).

Reply:
Conditions have no properties by definition. This is why we always have the 0 properties reserved for them.

The neighbours are computed once. I don't think this database access is a bottleneck.